cpp/src/io/parquet/reader_impl_chunking.cu

```diff
@@ -1232,22 +1232,17 @@ void reader::impl::setup_next_pass(read_mode mode)
       pass.skip_rows = _file_itm_data.global_skip_rows;
       pass.num_rows  = _file_itm_data.global_num_rows;
     } else {
-      auto const global_start_row = _file_itm_data.global_skip_rows;
-      auto const global_end_row   = global_start_row + _file_itm_data.global_num_rows;
       auto const start_row =
-        std::max(_file_itm_data.input_pass_start_row_count[_file_itm_data._current_input_pass],
-                 global_start_row);
+        _file_itm_data.input_pass_start_row_count[_file_itm_data._current_input_pass];
       auto const end_row =
         std::min(_file_itm_data.input_pass_start_row_count[_file_itm_data._current_input_pass + 1],
-                 global_end_row);
+                 _file_itm_data.global_num_rows);
```

Comment:
Really good question: This is because `_file_itm_data.input_pass_start_row_count` is computed from the selected row groups (in reader_impl_helpers.cpp) so the offsets are adjusted for `global_skip_rows` (meaning offset 0 = global_skip_rows). Thus we re-add `global_skip_rows` to `pass.start_row = start_row + global_skip_rows` but adding it to end_row is redundant since we only need to compute the `pass.num_rows = end_row + global_skip_rows - start_row - global_skip_rows`

Reply:
I hope this makes sense lol. Here's an example. Say global_skip_rows = 120; and we have `_file_itm_data.input_pass_start_row_count = [0, 80, 180, 280, 380]`, then we should have `pass.skip_rows = [120, 200, 300, 400,..] and `pass.num_rows = [80,100,100,100,100,..]`